src/overlays/actors/ovl_En_Elfgrp/z_en_elfgrp.c

```diff
@@ -27,54 +45,543 @@ const ActorInit En_Elfgrp_InitVars = {
+}
+
+s32 func_80A39C1C(GlobalContext* globalCtx, s32 arg1) {
+    // the permanentSceneFlags access here is in the form
+    // struct {
+    //   u32 clockTown : 1;
+    //   u32 fountains[4] : 5;
+    // } FairyFountains;
+    // where arg1 is:
+    //    0: clocktown
+    //    1: woodfall,
+    //    2: snowhead,
+    //    3: great bay,
+    //    4: stone tower
+    // clocktown is handled separately, and then the fountains are looked up as array indexing
+
+    s32 temp_v1;
+
+    if ((arg1 < 0) || (arg1 >= 5)) {
+        return 0;
+    }
+
+    if (arg1 == 0) {
+        if (gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 & 1) {
+            return 25;
+        }
+        return 24;
+    }
+
+    temp_v1 = (gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 >> (((arg1 - 1) * 5) + 1)) & 0x1F;
+    if (temp_v1 < 10) {
+        temp_v1 = 10;
+    } else if (temp_v1 > 25) {
+        temp_v1 = 25;
+    }
+    return temp_v1;
+}
+
+void func_80A39CD4(GlobalContext* globalCtx, s32 arg1, s32 arg2) {
+    if ((arg1 < 0) || (arg1 > 4) || (arg2 < 10) || (arg2 > 25)) {
+        return;
+    }
+
+    if (arg1 == 0) {
+        if (arg2 == 25) {
+            gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 |= 1;
+        } else {
+            gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 &= ~1;
+        }
+    } else {
+        gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 &= ~(0x1F << ((arg1 * 5) - 4));
+        gSaveContext.permanentSceneFlags[globalCtx->sceneNum].unk_14 |= arg2 << ((arg1 * 5) - 4);
+    }
+}
+
+void func_80A39DC8(EnElfgrp* this, GlobalContext* globalCtx, s32 arg2, s32 arg3) {
+    s32 pad;
+    s32 i;
+    Actor* elforg;
+    s32 temp;
+    Vec3f sp6C;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (arg3 == 0) {
+        this->unk_14A |= 8;
+    }
+
+    if (arg3 == 0) {
+        sp6C = this->actor.world.pos;
+        sp6C.y += 20.0f;
+        temp = ((this->unk_147 & 7) << 6) | 1;
+    } else {
+        sp6C = player->actor.world.pos;
+        sp6C.y += 20.0f;
+        temp = ((this->unk_147 & 7) << 6) | 8;
+    }
+
+    for (i = 0; i < arg2; i++) {
+        elforg =
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ELFORG, randPlusMinusPoint5Scaled(20.0f) + sp6C.x,
+                        sp6C.y, randPlusMinusPoint5Scaled(20.0f) + sp6C.z, 0, 0, 0, temp);
+        if (elforg == NULL) {
+            continue;
+        }
+        elforg->home.pos.x = this->actor.home.pos.x;
+        elforg->home.pos.y = this->actor.home.pos.y + 20.0f;
+        elforg->home.pos.z = this->actor.home.pos.z;
+    }
+}
+
+s32 func_80A39F50(GlobalContext* globalCtx) {
+    Actor* itemAction = globalCtx->actorCtx.actorLists[ACTORCAT_ITEMACTION].first;
+    EnElforg* elfOrg;
+
+    while (itemAction != NULL) {
+        if ((itemAction->id != ACTOR_EN_ELFORG) ||
+            ((ENELFGRP_GET(itemAction) != ENELFGRP_1) && (ENELFGRP_GET(itemAction) != ENELFGRP_8))) {
```

Comment:
This is actually applying to the elforg, so you can use the elforg macro/enum:
```suggestion
            ((STRAY_FAIRY_TYPE(itemAction) != STRAY_FAIRY_TYPE_FAIRY_FOUNTAIN) && (ENELFGRP_GET(itemAction) != STRAY_FAIRY_TYPE_TURN_IN_TO_FAIRY_FOUNTAIN))) {
```